src/lib/rest/StringFilter.cpp

```diff
@@ -920,28 +926,20 @@ bool StringFilter::parse(const char* q, std::string* errorStringP)
       return false;
     }
 
-    StringFilterItem item;
+    StringFilterItem* item = new StringFilterItem();
 
-    if (item.parse(s, errorStringP) == true)
+    if (item->parse(s, errorStringP) == true)
     {
       filters.push_back(item);
     }
     else
     {
+      free(item);
```

Comment:
You are right... I copied free() for the line below, but delete must be use. We alredy know that scaring things may happen when new/delete is mixxed with malloc/free :)